lucet-module-data/src/module.rs

```diff
@@ -0,0 +1,28 @@
+use crate::functions::FunctionSpec;
+use crate::module_data::ModuleData;
+use crate::tables::TableElement;
+
+pub const LUCET_MODULE_SYM: &str = "lucet_module";
+
+/// Module is the exposed structure that contains all the data backing a Lucet-compiled object.
+#[derive(Debug)]
+pub struct Module<'a> {
```

Comment:
:bike: :derelict_house: I anticipate confusion calling this a `Module`, given how pervasively we use that trait name in the runtime. Naturally, I don't have any better ideas off the top of my head :slightly_frowning_face: Maybe something with Lucet in the name to match the symbol?

Reply:
Thinking this through a bit:

On the `lucet-runtime` side, this is used only in `DlModule`, and not public, so it won't leak outside that file. There, I think explicitly naming `lucet_module_data::Module` seems ok to disambiguate.

Outside that use in `lucet-runtime`, I'm not sure it ends up being too confusing, because `lucet_module_data::Module` is only only in `lucet-analyze` at the moment and I don't really expect it to come up elsewhere.

Of course, maybe this reads very differently to someone new to the codebase :(

(One thought I did have and discard was renaming `lucet_runtime_internal::Module` to `lucet_runtime_internal::Executable`, with the name idea being that the runtime can turn an executable into an instance, and an executable is backed by a module or perhaps some other data. I'm not sold on that, but it's the best alternative I could think up)

Reply:
I think "executable" too strongly implies "a program with `main`", but I do like the thought.

And yeah, my main concern is confusion for people who are new to the codebase, but I'm fine with leaving it for now if it's really used in such a limited fashion.